### ISSUE TITLE
Cannot Start Debugging when package is directly under gopath

### ISSUE THREAD

**Issue by Ritzpak** [Microsoft/vscode-go#1765]
What version of Delve are you using (dlv version)?
5cd86b1
What version of Go are you using? (go version)?
go1.10.3.windows-amd64
What operating system and processor architecture are you using?
Windows Server 2016 Datacenter x64
Visual Studio Code v1.24.1 x64
What did you do?
Followed the install instructions.
What did you expect to see?
Hitting a breakpoint.
What did you see instead?
Error message: Can not debug non-main package.

I can Start Without Debugging and get my Println message so I know it compiles, but when I try Start Debugging, I get an error. I'm just getting started in Go and have only installed what is recommended so far. Not sure if there are conflicts or what. This should be extremely simple, but for some reason it won't run.  I can debug with commands from the terminal, but not through the IDE.

![image](https://user-images.githubusercontent.com/1278990/42161200-026d12d0-7dc8-11e8-872b-d602d8ae2755.png)
![image](https://user-images.githubusercontent.com/1278990/42161276-4b0712fc-7dc8-11e8-8b69-2cdbb6a5b76c.png)

**Comment by wiredprairie** [comment]
One idea: try launching the package rather than a single file. 

```
{
	"name": "Launch Package",
	"type": "go",
	"request": "launch",
	"mode": "debug",
	"program": "${workspaceRoot}"
}
```
Same error?

**Comment by ramya-rao-a** [comment]
> I can debug with commands from the terminal

Can you share a screenshot of that? How did you debug from the terminal? Using `dlv debug`? Did you pass a file path to it?

**Comment by ramya-rao-a** [comment]
Never mind, I got it to repro.

In this case `dlv debug` works but not `dlv debug testing`  and this only happens for the folder directly under the GOPATH. Try making another folder under your `testing` folder, add a go file there. That one will debug just fine.

This is an upstream issue. I have logged https://github.com/derekparker/delve/issues/1268

For now, try creating a sub folder and work there.
